src/main/kotlin/org/rust/cargo/project/workspace/StandardLibrary.kt

```diff
@@ -273,22 +277,32 @@ private class StdlibDataFetcher private constructor(
             project: Project,
             cargo: Cargo,
             srcDir: VirtualFile,
-            testPackageSrcDir: VirtualFile
+            testPackageSrcDir: VirtualFile,
+            version: RustcVersion
         ): Path? {
             val stdlibDependenciesDir = srcDir.pathAsPath.resolve("../vendor").normalize()
+            if (stdlibDependenciesDir.exists()) return stdlibDependenciesDir
 
-            if (!stdlibDependenciesDir.exists()) {
+            val stdlibHash = stdlibHash(srcDir, version)
+
+            val stdlibVendor = pluginDirInSystem().resolve("stdlib/${version.semver.parsedVersion}-$stdlibHash/vendor")
```

Comment:
Just a note: this can bloat IDEA system directory if a user updates its nightly toolchain every day

Reply:
I can't come up with anything better than some scheduled task to clean up old vendor dirs and some heuristic to check if particular vendor dir is actually "old"